statsforecast/core.py

```diff
@@ -117,12 +118,13 @@ def _cv_dates(last_dates, freq, h, test_size):
         else:
             total_dates = pd.date_range(end=last_dates[0], periods=test_size, freq=freq)
             out = np.empty((h * n_windows, 2), dtype='datetime64[s]')
-        for i_window in range(n_windows):
-            out[h * i_window : h * (i_window + 1), 0] = total_dates[i_window:(i_window + h)]
-            out[h * i_window : h * (i_window + 1), 1] = np.tile(total_dates[i_window] - freq * 1, h)
+        for i_window, cutoff in enumerate(range(-test_size, -h + 1, step_size), start=0):
+            end_cutoff = cutoff + h
+            out[h * i_window : h * (i_window + 1), 0] = total_dates[cutoff:] if end_cutoff == 0 else total_dates[cutoff:end_cutoff]
```

Comment:
Would be good to add comments here to help power users the thought process behind the construction of the dates and the forecast creation dates.